scripts/release/kubectl_mongodb/promote_kubectl_plugin.py

```diff
@@ -0,0 +1,304 @@
+    try:
+        subprocess.run(["git", "fetch", "--tags"], capture_output=True, text=True, check=True)
+
+        result = subprocess.run(
+            # using --short because that's how staging version is figured out for staging build scenario
+            # https://github.com/mongodb/mongodb-kubernetes/blob/1.5.0/scripts/dev/contexts/evg-private-context#L137
+            ["git", "rev-parse", "--short", f"{tag}^{{commit}}"],  # git rev-parse v1.1.1^{commit}
+            capture_output=True,
+            text=True,
+            check=True,
+        )
+        return result.stdout.strip()
+
+    except subprocess.CalledProcessError as e:
+        logger.info(f"Failed to get commit for tag: {tag}, err: {e.stderr.strip()}")
+        sys.exit(1)
+
+
+# generate_checksums generates checksums for the artifacts that we are going to upload to GitHub release as assets.
+# It's formatted: checksum  artifact_name
+def generate_checksums(artifacts_local_dir: str, artifacts: list[str]) -> str:
+    checksums_path = Path(os.path.join(artifacts_local_dir, "checksums.txt"))
+
+    with checksums_path.open("w") as out_file:
+        for artifact in artifacts:
+            artifact_path = Path(artifact)
+            if not artifact_path.is_file() or not artifact_path.name.endswith(".tar.gz"):
+                logger.info(f"skipping invalid tar file: {artifact_path}")
+                continue
+
+            sha256 = hashlib.sha256()
+            with open(artifact_path, "rb") as f:
+                # read chunk of 8192 bites until end of file (b"") is received
+                for chunk in iter(lambda: f.read(8192), b""):
+                    sha256.update(chunk)
+
+            checksum_line = f"{sha256.hexdigest()}  {artifact_path.name}"
+            out_file.write(checksum_line + "\n")
+
+    logger.info(f"Checksums written to {checksums_path}")
+    return str(checksums_path.resolve())
+
+
+# promote_artifacts promotes (copies) the downloaded staging artifacts to release S3 bucket.
+def promote_artifacts_to_s3(artifacts: dict[str, str], release_version: str, release_scenario_bucket_name: str):
+    s3_client = create_s3_client()
+
+    for local_file_path, s3_file_name in artifacts.items():
+        if not is_expected_artifact_for_promotion(local_file_path):
+            logger.warning(f"Skipping invalid or non-tar/checksum artifact: {local_file_path}")
+            continue
+
+        s3_key = s3_path(s3_file_name, release_version)
+
+        try:
+            s3_client.upload_file(local_file_path, release_scenario_bucket_name, s3_key)
+            logger.debug(f"{local_file_path} was promoted to s3://{release_scenario_bucket_name}/{s3_key} successfully")
+        except ClientError as e:
+            raise Exception(f"failed to upload the file {local_file_path}: {e}")
+
+    logger.info("All artifacts were promoted to release bucket successfully")
+
+
+def is_expected_artifact_for_promotion(file_path: str) -> bool:
```

Comment:
is this required because now we have the compressed as well as decompressed file in staging? I am a little sceptical about the idea because it would make things confusing, like we discussed on call.
But if we HAVE to do this, we should add a comment here explaining that.

Reply:
Would it be problematic if run `tar` locally and don't upload copies or artifacts in S3.

Reply:
>is this required because now we have the compressed as well as decompressed file in staging?

only in release bucket, in staging we still have only decompressed ones.

>if we HAVE to do this, we should add a comment here explaining that

yes, we have to put both compressed as well as decompressed files. Compressed are used for release assets backup while uncompressed are used for e2e testing. I wanted to keep the simple logic when calling `download_multi_cluster_binary` that is used when building test image and not having to decompress binaries, when downloading from release bucket. It's much simpler to just push both flavours.